lib/types/index.ts

```diff
@@ -13,3 +13,9 @@ export type { PrState } from './pr-state';
 export type { ModuleApi, RenovatePackageJson } from './base';
 
 export type AutoMergeType = 'branch' | 'pr' | 'pr-comment';
+
+type Val = NonNullable<unknown>;
+
+export type Nullable<T extends Val> = T | null | undefined;
```

Comment:
It's somewhat counterintuitive that besides null, this type may also be undefined. There are plenty of cases we're changing here, where we'll now allow extra values that we didn't allow before.